src/include/miopen/solver.hpp

```diff
@@ -2672,12 +2697,9 @@ struct PerformanceConfigAsmImplicitGemmGTC : Serializable<PerformanceConfigAsmIm
     template <class Self, class F>
     static void Visit(Self&& self, F f)
     {
-        std::string prec_string = self.precision == miopenFloat
-                                      ? "fp32"
-                                      : (self.precision == miopenHalf ? "fp16" : "bf16");
         f(self.direction, "dir");
         f(self.tensor_layout, "lyt");
-        f(prec_string, "pre");
+        f(self.precision, "pre");
```

Comment:
This the most important fix. Adds deserialization for precision, which was missing and that was the root of the problems.